src/modules/peek/Peek.UI/Helpers/FileExplorerHelper.cs

```diff
@@ -2,27 +2,36 @@
 // The Microsoft Corporation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Collections.Generic;
-using System.Diagnostics;
-using Peek.Common.Models;
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
 using Peek.UI.Native;
+using Windows.Win32;
+using Windows.Win32.Foundation;
 
 namespace Peek.UI.Helpers
 {
     public static class FileExplorerHelper
     {
-        public static Shell32.IShellFolderViewDual2? GetCurrentFolderView()
+        public static unsafe Shell32.IShellFolderViewDual2? GetCurrentFolderView()
         {
             var foregroundWindowHandle = NativeMethods.GetForegroundWindow();
 
+            int capacity = PInvoke.GetWindowTextLength(new HWND(foregroundWindowHandle)) * 2;
+            StringBuilder foregroundWindowTitleBuffer = new StringBuilder(capacity);
+            NativeMethods.GetWindowText(new HWND(foregroundWindowHandle), foregroundWindowTitleBuffer, foregroundWindowTitleBuffer.Capacity);
+
+            string foregroundWindowTitle = foregroundWindowTitleBuffer.ToString();
```

Comment:
If we have two tabs open with the same name, this could get the wrong tab right? Since I assume this can only happen if it's the same folder opened twice, it's only problematic if selected items or the sort option is different between them.

Reply:
Either way, this will work a lot better than before, and we can tackle always those edge cases in the future

Reply:
Ah, yeah you're right. Good point... I guess we can get that in as that will fix the issue for the majority of cases, and we can keep working on fixing that edge case.